thread_pthread.c

```diff
@@ -1765,10 +1765,23 @@ native_thread_native_thread_id(rb_thread_t *target_th)
     if (tid == 0) return Qnil;
     return INT2FIX(tid);
 #elif defined(__APPLE__)
-    uint64_t tid;
-    int e = pthread_threadid_np(target_th->nt->thread_id, &tid);
-    if (e != 0) rb_syserr_fail(e, "pthread_threadid_np");
-    return ULL2NUM((unsigned long long)tid);
+    #if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
+        uint64_t tid;
+        tid = pthread_mach_thread_np(pthread_self());
+    #elif MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+        uint64_t tid;
+        if (&pthread_threadid_np) {
+            int e = pthread_threadid_np(target_th->thread_id, &tid);
```

Comment:
Please prefer `error` or `error_number` over `e`.

Reply:
@ioquatix Oh sure, it will be certainly more readable. I just followed what was there in the source (and it just used `e`).